File: 268.missing-number.ts
/*
 * @lc app=leetcode id=268 lang=typescript
 *
 * [268] Missing Number
 */

// @lc code=start
function missingNumber(nums: number[]): number {
   let missing = nums.length; // Initialize missing to n
   for (let i = 0; i < nums.length; i++) {
       missing ^= i ^ nums[i];
   }
   return missing;
};
// @lc code=end

File: 268.missing-number.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#

# @lc code=start
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        n = len(nums)
        totalSum = n*(n+1)//2
        sum = 0
        for num in nums:
            sum += num
        return totalSum - sum
# @lc code=end

